src/classes/OPP_CampaignMember_TDTM.cls

```diff
@@ -0,0 +1,126 @@
+    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist, 
+        TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+        
+        dmlWrapper dmlWrapper = new DmlWrapper();
+
+        if (hasRun)
+            return dmlWrapper;
+            
+        for (integer i=0; i<newlist.size(); i++) {
+
+            Opportunity opp = (Opportunity)newlist[i];
+            
+            //if status is "omit", or the primary contact or campaign are blank,
+            //continue to the next opportunity without doing any campaign member work
+            if ((!string.isBlank(opp.Primary_Contact_Campaign_Member_Status__c) && 
+                    opp.Primary_Contact_Campaign_Member_Status__c.equalsIgnoreCase(Label.CampaignMemberStatusOmit)) ||
+                    opp.Primary_Contact__c == null || opp.CampaignId == null) {
+
+                continue;
+            }
+
+            if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
+                addOppMemberFields(opp);
+            }
+            
+            if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {
+                Opportunity oldOpp = (Opportunity)oldlist[i];
+
+                //the status is blank and we're changing to or from closed/won
+                if (string.isBlank(opp.Primary_Contact_Campaign_Member_Status__c) && 
+                    (opp.IsClosed != oldOpp.isClosed && opp.IsWon != oldOpp.isWon)) {
+                    
+                    addOppMemberFields(opp);
+                //the primary contact, campaign, or status changed
+                } else if (opp.Primary_Contact__c != oldOpp.Primary_Contact__c || 
+                    opp.CampaignId != oldOpp.CampaignId || 
+                    opp.Primary_Contact_Campaign_Member_Status__c != oldOpp.Primary_Contact_Campaign_Member_Status__c) {
+                    
+                    addOppMemberFields(opp);
+                }
+            }
+        }
+        
+        if (!listContactId.isEmpty())
+            dmlWrapper = CON_AddToCampaign.CreateCampaignMembers(listContactId, listCampaignId, listMemberStatus);
+
+        hasRun = true;
+        return dmlWrapper;
+    }
+
+    /*******************************************************************************************************
+    * @description Adds a Contact Id, Campaign Id, and Status to our three lists. If the opportunity has a
+    * status it is used, otherwise a default is used based on if the opportunity is closed/won or not.
+    * @param Opp The current opportunity.
+    * @return void This method adds to three lists and returns nothing.
+    ********************************************************************************************************/
+    private void addOppMemberFields(Opportunity opp) {
+        //if no status is provided, default to a responded status if the opp is closed/won
```

Comment:
I'm filtering them out in the first if block line 70. I'll change it to make it more legible.